### PR TITLE
Add Serilog (logging library) and components

### PR DESCRIPTION
**Pull request by schultzcole**
> The NuGet package needs to respect a few constraints in order to be listed in the curated list:
> - [X] It must have non-preview versions (e.g 1.0.0 but not 1.0.0-preview.1)
> - [X] It must provide .NETStandard2.0 assemblies as part of its package
> - [X] The lowest version added must be the lowest .NETStandard2.0 version available
> - [X] The package has been tested with the Unity editor 
> - [X] The package has been tested with a Unity standalone player
>   - if the package is not compatible with standalone player, please add a comment to a Known issues section to the top level readme.md
> - [X] All package dependencies with .NETStandard 2.0 target must be added to the PR (respecting the same rules above)
>   - Note that if a future version of the package adds a new dependency, this dependency will have to be added manually as well

### CODE REVIEW

**Comment by bdovaz** [comment]
@schultzcole Please, can you update this PR and add this other packages?

Microsoft.Extensions.* integrations (this packages are already included in registry.json):

https://www.nuget.org/packages/Serilog.Settings.Configuration/
https://www.nuget.org/packages/Serilog.Extensions.Hosting/
https://www.nuget.org/packages/Serilog.Extensions.Logging/

Thanks!

File: registry.json [hunk 1]
```diff
@@ -230,6 +230,30 @@
     "listed": true,
     "version": "2.1.0"
   },
+  "Serilog": {
+    "listed": true,
+    "version": "2.8.0"
+  }
```
Comment (xoofx) [inline registry.json:236]: missing `,` after each entry.

Comment (schultzcole) [reply registry.json]: Ah, sorry about that. That should be fixed now.

**Comment by schultzcole** [comment]
> https://www.nuget.org/packages/Serilog.Settings.Configuration/
> https://www.nuget.org/packages/Serilog.Extensions.Hosting/
> https://www.nuget.org/packages/Serilog.Extensions.Logging/

@bdovaz these packages seem to be related to ASP.NET core, I'm not sure they're relevant to Unity

**Comment by bdovaz** [comment]
@schultzcole Why do you say that? It has no dependencies other than to Serilog or Microsoft.Extensions.* packages. It has nothing to do with ASP.

If you are talking about appsettings.json files, it is a pattern that can be perfectly used in Unity, we do it.

**Comment by bdovaz** [comment]
@schultzcole After making this change in the registry.json file have you tried running the server? There are missing dependencies.

Is there a way that in CI you can check @xoofx this fact? Because it is easy to miss a dependency as in this case and hide the problem.

image

**But before you look at it**, read on because I've discovered several things.

I have made a fork to try to add the 3 packages I mentioned.

Along the way I have encountered several things.

1. The minimum version of Serilog that you have put is 2.8.0 but in spite of targeting netstandard2.0 it depends in turn on another package (System.Collections.NonGeneric) that only targets netstandard1.3.

That is why in this case you should put as minimum version 2.9.0 instead of 2.8.0.

2. Serilog.Enrichers.Thread: Depends on Serilog >= 2.3.0 so it is not possible to add it since these versions did not support netstandard2.0.

3. Serilog.Formatting.Compact: Depends on Serilog >= 2.8.0 so it is not possible to add it because as I mentioned in point 1 it is not possible to add this version.

4. Serilog.Sinks.Async: the same as point 3.

5. Serilog.Sinks.File: Depends on Serilog >= 2.5.0 with which it is not possible to add it since these versions did not support netstandard2.0.

5. Serilog.Sinks.File.Header: As it depends on Serilog.Sinks.File it has the problem derived from this one.

As for the dependencies that I proposed I have seen the following:

Serilog.Extensions.Hosting
Serilog.Extensions.Logging

They depend on a PR I just made of the Logging package that lifts from 2.8 to 2.9 and would make it compatible:

https://github.com/serilog/serilog-extensions-logging/pull/188

Serilog.Settings.Configuration

Depends on a stable version 3.2.0 which is the one that starts to depend on a version >= 2.10.0

**In short**, to support the packages that extend Serilog that you propose, it would be necessary to make several PRs similar to the one I have made so that they can upgrade the minimum version of Serilog to at least 2.9.0.

Serilog.Enrichers.Thread
Serilog.Formatting.Compact
Serilog.Sinks.Async
Serilog.Sinks.File
Serilog.Sinks.File.Header

**Comment by schultzcole** [comment]
@bdovaz Ah, thanks for looking in to that. I guess this PR will have to wait on the back burner until all of these libraries drop support for serilog versions < 2.9.0.

Maybe just add Serilog 2.9.0 for now, and I could make separate PRs for additional libraries as they become compatible.

**Comment by KuraiAndras** [comment]
Any update on this?

**Comment by bdovaz** [comment]
Hi, I've created a PR #51 with the following:

Serilog
Serilog.Settings.Configuration
Serilog.Sinks.Async
Serilog.Sinks.Console
Serilog.Sinks.Debug
Serilog.Sinks.File
Serilog.Sinks.Trace

There would still be some left:

Serilog.Enrichers.Thread
Serilog.Formatting.Compact
Serilog.Sinks.File.Header

These cannot yet be added because they have a Serilog dependency lower than 2.9.0 and the dependency chain is not compatible with .NET Standard 2.0.

So as you see @schultzcole, either you close your PR or we keep insisting to the author on a release of a stable version of those packages:

https://www.nuget.org/packages/Serilog.Enrichers.Thread/3.2.0-dev-00752

https://www.nuget.org/packages/Serilog.Formatting.Compact/1.1.1-dev-00944

https://www.nuget.org/packages/Serilog.Sinks.File.Header/ (For this one we need a PR first to release the version to >= 2.9.0)

To notify the author, there is this channel:

https://gitter.im/serilog/serilog
